Source/WebCore/rendering/RenderSearchField.cpp

```diff
@@ -202,7 +202,7 @@ Visibility RenderSearchField::visibilityForCancelButton() const
 
 const AtomString& RenderSearchField::autosaveName() const
 {
-    return inputElement().attributeWithoutSynchronization(autosaveAttr);
+    return inputElement().attributeWithoutSynchronization(nameAttr);
```

Comment:
Can we create a test here? I don’t think that just changing this with no test coverage is a good idea.

I understand this is not a standard and unprecedented in other web browsers. But is there any app or website relying on this behavior in WebKit? We need to quickly research that before removing any feature.

Reply:
Normally we require regression tests for any change to WebKit. Or we explain why this needs to be an exception to that rule.

Reply:
> Can we create a test here? I don’t think that just changing this with no test coverage is a good idea.

I added a test that checks the saved items via `internals.recentSearches`. It's indeed helpful. With it I found the items are not saved to disk because we didn't create parent directory of the plist file.

> I understand this is not a standard and unprecedented in other web browsers. But is there any app or website relying on this behavior in WebKit? We need to quickly research that before removing any feature.

I checked existing tests, and there's only one old imported test uses "autosave" attribute, and does not check if the item is actually saved. Based on the facts that
1. we don't have any test coverage
2. it's not documented anywhere (I checked standard and [mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search))
3. it might be broken for a while (as we didn't ensure directory exists)
The chance that app or website relying on this might be small. I think we could just go ahead and do this for compatibility, and watch out for breakage.